ibrcommon/tests/ssl/CipherStreamTest.cpp

```diff
@@ -279,16 +280,17 @@ void CipherStreamTest::aesstream_test01()
 
 	// encrypt the test data
 	{
-		ibrcommon::AES128Stream crypt_stream(ibrcommon::CipherStream::CIPHER_ENCRYPT, data, key, salt);
+		refcnt_ptr<ibrcommon::AES128Stream> crypt_stream = refcnt_ptr<ibrcommon::AES128Stream>(
+				new ibrcommon::AES128Stream(ibrcommon::CipherStream::CIPHER_ENCRYPT, data, key, salt));
```

Comment:
Is there a reason why you explicitly instantiate and assign the object instead of a simple initialization?

```
refcnt_ptr<ibrcommon::AES128Stream> crypt_stream(
    new ibrcommon::AES128Stream(ibrcommon::CipherStream::CIPHER_ENCRYPT, data, key, salt));
```

Reply:
No reason other than writing Java and C++ code on the same day...